### PR TITLE
txn:improvement for point get by unique key or pk

### PR DESCRIPTION
**Pull request by AndreMouche**
This PR implement improvement for point get by unique key or pk with tidb.
For `Get` with ts==uint64::max(which means it's a point get by pk or unique_key from tidb),  if  lock exist and current key the primary, return latest committed version instead of ErrorWithLock.  
@disksing @zhangjinpeng1987 PTAL

### CODE REVIEW

**Comment by siddontang** [comment]
PTAL @hhkbp2

File: src/storage/mvcc/reader.rs [hunk 1]
```diff
@@ -176,13 +177,22 @@ impl<'a> MvccReader<'a> {
...
         if let Some(lock) = try!(self.load_lock(key)) {
             if lock.ts <= ts {
-                // There is a pending lock. Client should wait or clean it.
-                return Err(Error::KeyIsLocked {
-                    key: try!(key.raw()),
-                    primary: lock.primary,
-                    ts: lock.ts,
-                    ttl: lock.ttl,
-                });
+                let raw_key = try!(key.raw());
```
Comment (disksing) [inline src/storage/mvcc/reader.rs]: Move it into `if ts == u64::MAX`.

File: src/storage/mvcc/reader.rs [hunk 2]
```diff
@@ -176,13 +177,22 @@ impl<'a> MvccReader<'a> {
...
+                let primary_key = lock.primary;
+                if ts == u64::MAX && raw_key == primary_key {
+                    // when ts==MAX(which means it's a point get by pk or unique key),
+                    // and current key is the primary key,
+                    // return latest commit version's value
```
Comment (disksing) [inline src/storage/mvcc/reader.rs]: It looks strange for MVCC to be mixed with the concept of _pk_ or _unique key_. Maybe better to extract a `get_latest(&mut self, key: &Key)` method.

**Comment by ngaut** [comment]
PTAL

File: src/storage/mvcc/reader.rs [hunk 3]
```diff
@@ -176,13 +177,20 @@ impl<'a> MvccReader<'a> {
...
             if lock.ts <= ts {
-                // There is a pending lock. Client should wait or clean it.
-                return Err(Error::KeyIsLocked {
-                    key: try!(key.raw()),
-                    primary: lock.primary,
-                    ts: lock.ts,
-                    ttl: lock.ttl,
-                });
+                if ts == u64::MAX && try!(key.raw()) == lock.primary {
+                    // when ts==MAX(which means get latest committed version for primary key),
```
Comment (hhkbp2) [inline src/storage/mvcc/reader.rs]: s/ts== MAX(which means get / ts == u64::MAX (which means to get/

**Comment by hhkbp2** [comment]
Rest LGTM

**Comment by disksing** [comment]
LGTM

**Comment by ngaut** [comment]
PTAL @hhkbp2

**Comment by siddontang** [comment]
any performance increased?

**Pull request merged**